### ISSUE TITLE
aws_service_discovery_private_dns_namespace name length limit

### ISSUE THREAD

**Issue by ahl** [terraform-providers/terraform-provider-aws#4701]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.20.0

### Affected Resource(s)

aws_service_discovery_private_dns_namespace

### Terraform Configuration Files

```hcl
provider "aws" {
  region  = "us-west-2"
  version = "1.20.0"
}

resource "aws_service_discovery_private_dns_namespace" "fail" {
  name = "this.is.a.really.long.name.i.suppose.but.who.are.you.to.judge"
  vpc  = "vpc-1234"
}

```

### Debug Output

https://gist.github.com/ahl/db1f84f4df6669fec9215cc0b84add10

### Expected Behavior

Should have created the resource.

### Actual Behavior

failed:
```
* aws_service_discovery_private_dns_namespace.fail: InvalidInput: 1 validation error detected: Value 'tf-this.is.a.really.long.name.i.suppose.but.who.are.you.to.judge20180530163701323000000001' at 'creatorRequestId' failed to satisfy constraint: Member must have length less than or equal to 64
```

### Steps to Reproduce

1. `terraform apply`

**Comment by bflad** [comment]
The fix will be released with v1.22.0 of the AWS provider, middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.22.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1220-june-05-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
